modules/influxdb/influxdb.go

```diff
@@ -0,0 +1,144 @@
+			// terminate the influxdb server, followed by restart of influxdb.  This is tricky to wait for, and
+			// in this case, we are assuming that data was added by init script, so we then look for an
+			// "Open shard" which is the last thing that happens before the server is ready to accept connections.
+			// This is probably different for InfluxDB 2.x, but that is left as an exercise for the reader.
+			genericContainerReq.WaitingFor = wait.ForAll(
+				wait.ForLog("influxdb init process in progress..."),
+				wait.ForLog("Server shutdown completed"),
+				wait.ForLog("Opened shard"),
+				wait.ForExposedPort())
+			hasInitDb = true
+			break
+		}
+	}
+	if !hasInitDb {
+		if lastIndex := strings.LastIndex(genericContainerReq.Image, ":"); lastIndex != -1 {
+			tag := genericContainerReq.Image[lastIndex+1:]
+			if tag == "latest" || tag[0] == '2' {
+				genericContainerReq.WaitingFor = wait.ForLog(`Listening log_id=[0-9a-zA-Z_]+ service=tcp-listener transport=http`).AsRegexp()
+			}
+		} else {
+			genericContainerReq.WaitingFor = wait.ForLog("Listening for signals")
+		}
+	}
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	if err != nil {
+		return nil, err
+	}
+	return &InfluxDbContainer{container}, nil
+}
+
+func (c *InfluxDbContainer) MustHaveConnectionUrl(ctx context.Context, tls bool) string {
+	connectionString, err := c.ConnectionUrl(ctx, tls)
+	if err != nil {
+		panic(err)
+	}
+	return connectionString
+}
+
+func (c *InfluxDbContainer) ConnectionUrl(ctx context.Context, tls bool) (string, error) {
+	containerPort, err := c.MappedPort(ctx, "8086/tcp")
+	if err != nil {
+		return "", err
+	}
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+	var connectionString string
+	if tls {
+		connectionString = fmt.Sprintf("https://%s:%s", host, containerPort.Port())
+	} else {
+		connectionString = fmt.Sprintf("http://%s:%s", host, containerPort.Port())
+	}
+	return connectionString, nil
+}
+
+func WithUsername(username string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["INFLUXDB_USER"] = username
+	}
+}
+
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["INFLUXDB_PASSWORD"] = password
+	}
+}
+
+func WithDatabase(database string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["INFLUXDB_DATABASE"] = database
+	}
+}
+
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: "/etc/influxdb/influxdb.conf",
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+	}
+}
+
+// WithInitDb will copy a 'docker-entrypoint-initdb.d' directory to the container.
```

Comment:
I think we cannot enforce users to create a `docker-entrypoint-initdb.d` dir into their projects, but let them know that the dir passed to this function will be copied to the container with that name.

I'd advocate for hardcoding `/docker-entrypoint-initdb.d` in https://github.com/testcontainers/testcontainers-go/pull/1703/files#diff-328ce3ff8f6404a82f6ade95861e87cb8e19ef7d45dd4579fad562d8dfde4f7fR139, removing the `docker-entrypoint-initdb.d` suffix from https://github.com/testcontainers/testcontainers-go/pull/1703/files#diff-328ce3ff8f6404a82f6ade95861e87cb8e19ef7d45dd4579fad562d8dfde4f7fR138

Reply:
After this time, I ended up doing exactly the same as in the original PR. SO let's keep it simple by the moment, and improve it if needed.